common/taskvalidator/validateworkflow.go

```diff
@@ -49,14 +56,34 @@ func NewWfChecker(logger log.Logger, metrics metrics.Client) Checker {
 // WorkflowCheckforValidation is a dummy implementation of workflow validation.
 func (w *checkerImpl) WorkflowCheckforValidation(workflowID string, domainID string, domainName string, runID string) error {
 	// Emitting just the log to ensure that the workflow is called for now.
-	// TODO: add some validations to check the wf for corruptions.
+	// TODO: Ass tsale workflow check validation.
 	w.logger.Info("WorkflowCheckforValidation",
 		tag.WorkflowID(workflowID),
 		tag.WorkflowRunID(runID),
 		tag.WorkflowDomainID(domainID),
 		tag.WorkflowDomainName(domainName))
 	// Emit the number of workflows that have come in for the validation. Including the domain tag.
 	// The domain name will be useful when I introduce a flipr switch to turn on validation.
+	// TODO: Add this as a first validation before the stale workflow check. Add a deleteworkflow call if true.
+	err := w.deprecatedDomainCheck(domainID, domainName)
+	if err != nil {
+		return err
+	}
 	w.metricsClient.Scope(metrics.TaskValidatorScope, metrics.DomainTag(domainName)).IncCounter(metrics.ValidatedWorkflowCount)
 	return nil
 }
+
+func (w *checkerImpl) deprecatedDomainCheck(domainID string, domainName string) error {
+	domain, err := w.dc.GetDomainByID(domainID)
+	if err != nil {
+		w.logger.Error("Error getting domain by ID", tag.Error(err))
+		return err
```

Comment:
in general it is better to avoid logging the error and also returning it. ideally somewhere close to the top of the call chain does the error logging. e.g. a middleware in an rpc chain.
if you need additional information to be logged then you can wrap the error and return